pkg/cleanup/cleanup.go

```diff
@@ -62,6 +65,42 @@ func GetInactiveTags(activeTags, tags *[]string) []string {
 	return inactiveTags
 }
 
+// GetOrphanImageTags returns the tags that does not have any git commit match
+func GetOrphanImageTags(gitValues, imageTags *[]string, matchOption MatchOption) []string {
+	var orphans []string
+
+	log.WithField("gitValues", gitValues).Debug("Git commits/tags")
+	log.WithField("imageTags", imageTags).Debug("Image stream tags")
+
+	for _, tag := range *imageTags {
+		orphans = append(orphans, tag)
```

Comment:
Please change the logic to only add a tag to the orphans if no match was found (instead of adding it and removing again)